File: dss-portal-api/src/main/webapp/app/controllers/portal.js
/*
 * Copyright 2019 Infosys Ltd.
 *
 * Licensed to the Apache Software Foundation (ASF) under one or more
 * contributor license agreements.  See the NOTICE file distributed with
 * this work for additional information regarding copyright ownership.
 * The ASF licenses this file to You under the Apache License, Version 2.0
 * (the "License"); you may not use this file except in compliance with
 * the License.  You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
/**
 * Description : Controller for Logistics DSS.
 */

app.controller("PortalCtrl", ['$scope', '$rootScope','$route', '$http','$templateCache','uiGridConstants','$filter','mainService','$sce','mainFactory','ngDialog',function($scope, $rootScope,$route,$http,$templateCache,$uiGridConstants,$filter,mainService,$sce,mainFactory,ngDialog){
	
	$scope.dashboardNameLogo = false;
	$scope.portalSelected = false;	
	$scope.retrievedportalId = localStorage.getItem('portalId');
	$scope.retrievedroleId = localStorage.getItem('roleId');
	$scope.selectedRole = localStorage.getItem('roleName');
	

	$scope.getPortalDetails = function(portalID){
		var token = localStorage.getItem('loginToken');
		return mainService.getPortalDetails(portalID,token).then(
				function(data) {
					if (data.success == true) {
						$scope.portalRecord = data.root;
						$scope.portalName = $scope.portalRecord.strPortalName;
						$scope.cssFile = "";
						$scope.logoFile = "";
						$scope.cssFile.name = $scope.portalRecord.strPortalCss;
						$scope.logoFile.name = $scope.portalRecord.strPortalLogo;						
					} else {
						var error = data.errors;
						console.log(error.errorMessage);
						if(error.errorCode=="3006"){
							$scope.openConfirmLogin();
						}
					}
				});
	}
	
	$scope.getDashboardList = function(portalId){
		/*var toggleValue = document.getElementById("dashboardToggle");*/
		var roleId = $scope.retrievedroleId
		var token = localStorage.getItem('loginToken');
		return mainService.getExistingDashboardList(portalId,roleId,token).then(
				function(data) {
					if (data.success == true) {
						$scope.dashboardsList = data.root;
						console.log($scope.dashboardsList)
						$scope.existingDashboards = data.root;
						console.log($scope.existingDashboards);
						 angular.forEach($scope.existingDashboards, function(val){
							 var d = new Date(val.createdDate);
							 val.createdDate  = d.toDateString();
						 });
						console.log(JSON.stringify($scope.existingDashboards));
						var id = $scope.existingDashboards[0].dashboardId;
						var name = $scope.existingDashboards[0].strDashboardlName;
						$scope.dashboardClick(id,name);
					} else {
						var error = data.errors;
						$scope.existingDashboards = "";
						console.log(error.errorMessage);
						if(error.errorCode=="3006"){
							$scope.openConfirmLogin();
						}
					}
				});
	}
	
	var loginData = localStorage.getItem('loginObject');
	$scope.openConfirmLogin = function() {
		ngDialog.openConfirm({
			template : 'modalDialogLogin',
			className : 'ngdialog-theme-default'
		}).then(
				function(value) {
					//callback();
					window.location = 'login.html';	
				},
				function(reason) {
					console.log('Modal promise rejected. Reason: ', reason);
				});
	};	
	if(!loginData){
		$scope.openConfirmLogin();
	}
	else{
		$scope.loginObject = localStorage.getItem('loginObject');
		$scope.loginObject = JSON.parse($scope.loginObject);
		$scope.userName = $scope.loginObject.userSession.userName;
		$scope.getDashboardList($scope.retrievedportalId);
		$scope.getPortalDetails($scope.retrievedportalId);
		
	}
	
	/*$scope.$on("CallMethod", function (event, args) {
		var portalId = args.portalId;
		$scope.getDashboardList(portalId);
	});*/
	
	/*$rootScope.$on("CallMethod", function(event, args){
        $scope.getDashboardList(portalId);
	});*/
	
	
	

	
	/*$scope.getPortalId = function(){
		var portalID = mainFactory.getPortalValue();
		$scope.getDashboardList(portalID);
	};*/
	
	
	$scope.dashboardClick = function(dashboardId,dashboardName){
		var token = localStorage.getItem('loginToken');
		var roleId = $scope.retrievedroleId
		$scope.dashboardName = dashboardName;
		$scope.dashboardNameLogo = true;
		$scope.dashboardNameClass = "active";
		$scope.activeDashboard = dashboardId;
		
		return mainService.getExistingCategoriesList(dashboardId,roleId,token).then(
				function(data) {
					if (data.success == true) {
						$scope.existingCategoriesList = data.root;
						console.log(JSON.stringify($scope.existingCategories));
						/*var kpisList = $scope.existingCategoriesList[0].kpis;
						var visList = kpisList[0].visualizations;
						var vis = visList[0];
						$scope.viewVisualization(visList)*/
						$scope.visualizationDisplayList = [];
						angular.forEach($scope.existingCategoriesList, function(val){
							if(val.kpis){
								angular.forEach(val.kpis, function(val_1){
									if(val_1.visualizations){
										angular.forEach(val_1.visualizations, function(val_2){
											val_2.selected=true;
											$scope.viewVisualization(val_2);
										})
									}
								})
							}
						})
					} else {
						var error = data.errors;
						$scope.existingCategories = "";
						console.log(error.errorMessage);
						if(error.errorCode=="3006"){
							$scope.openConfirmLogin();
						}
					}
				});
	}
	
	$scope.viewVisualization = function(vis){
		/*$scope.activeMenu = true;*/
		if(vis.selected == true){
			var visualize = JSON.parse(vis.strVisualizeConfigDetails)
			$scope.graphUrl =  visualize.strSinkQuery;
			$scope.visualizationName = vis.strVisualizeName; 
			var graphObj = {
					"id"  : vis.inVisualizeId,
					"name" : $scope.visualizationName,
					"url" : $scope.graphUrl
			}
			if($scope.visualizationDisplayList.length < 4){
				$scope.visualizationDisplayList.push(graphObj);
			}
		}
		else if(vis.selected == false){
			var visualize = JSON.parse(vis.strVisualizeConfigDetails)
			$scope.graphUrl =  visualize.strSinkQuery;
			$scope.visualizationName = vis.strVisualizeName; 
			var index = -1;
			$scope.visualizationDisplayList.some(function(obj, i) {
			  return obj.id === vis.inVisualizeId ? index = i : false;
			});			
			$scope.visualizationDisplayList.splice(index,1);
		}
	}
	
	$scope.trustSrc = function(src) {
		return $sce.trustAsResourceUrl(src);
	};
	
	$scope.logout = function(){
		var token = localStorage.getItem('loginToken');
		return mainService.logout(token).then(
				function(data) {
					if (data.success == true) {
						$scope.logout = data.root;//response.data.root.userSession.uuid						
						if($scope.logout.statusCode == '3008'){
							alert($scope.logout.statusMessage);
						}else if($scope.logout.statusCode == '3012'){
							alert($scope.logout.statusMessage);
						}else if($scope.logout.statusCode == '3011'){
							 window.location = 'login.html';
							 localStorage.removeItem('loginToken');
							 var token = localStorage.getItem('loginToken');
						}
						console.log(JSON.stringify($scope.logout));
					} else {
						var error = data.errors;
						console.log(error.errorMessage);
					}
				});
	}
		 
		
}]);

File: dss-stream-api/src/main/java/org/streamconnect/dss/api/ui/response/Portal.java
/*
* Copyright 2019 Infosys Ltd.
*
* Licensed to the Apache Software Foundation (ASF) under one or more
* contributor license agreements.  See the NOTICE file distributed with
* this work for additional information regarding copyright ownership.
* The ASF licenses this file to You under the Apache License, Version 2.0
* (the "License"); you may not use this file except in compliance with
* the License.  You may obtain a copy of the License at
*
*    http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/
package org.streamconnect.dss.api.ui.response;

import java.io.Serializable;
import java.util.Date;
import java.util.List;

/**
 * The Class Portal.
 *
 * @version 1.0
 */
public class Portal implements Serializable {

    /** The portal id. */
    private int portalId;

    /** The str portal name. */
    private String strPortalName;

    /** The str portal title. */
    private String strPortalTitle;

    /** The str portal url. */
    private String strPortalUrl;

    /** The str portal logo. */
    private String strPortalLogo;

    /** The str portal css. */
    private String strPortalCss;

    /** The str created user. */
    private String strCreatedUser;

    /** The str updated user. */
    private String strUpdatedUser;

    /** The created date. */
    private Date createdDate;

    /** The updated date. */
    private Date updatedDate;

    /** The delete status. */
    private int deleteStatus;

    /** The dashboards. */
    private List<Dashboard> dashboards;

    /** The is portal view enabled. */
    private boolean isPortalViewEnabled;

    /**
     * Gets the portal id.
     *
     * @return the portal id
     */
    public int getPortalId() {
        return portalId;
    }

    /**
     * Sets the portal id.
     *
     * @param portalId
     *            the new portal id
     */
    public void setPortalId(final int portalId) {
        this.portalId = portalId;
    }

    /**
     * Gets the str portal name.
     *
     * @return the str portal name
     */
    public String getStrPortalName() {
        return strPortalName;
    }

    /**
     * Sets the str portal name.
     *
     * @param strPortalName
     *            the new str portal name
     */
    public void setStrPortalName(final String strPortalName) {
        this.strPortalName = strPortalName;
    }

    /**
     * Gets the str portal title.
     *
     * @return the str portal title
     */
    public String getStrPortalTitle() {
        return strPortalTitle;
    }

    /**
     * Sets the str portal title.
     *
     * @param strPortalTitle
     *            the new str portal title
     */
    public void setStrPortalTitle(final String strPortalTitle) {
        this.strPortalTitle = strPortalTitle;
    }

    /**
     * Gets the str portal url.
     *
     * @return the str portal url
     */
    public String getStrPortalUrl() {
        return strPortalUrl;
    }

    /**
     * Sets the str portal url.
     *
     * @param strPortalUrl
     *            the new str portal url
     */
    public void setStrPortalUrl(final String strPortalUrl) {
        this.strPortalUrl = strPortalUrl;
    }

    /**
     * Gets the str portal logo.
     *
     * @return the str portal logo
     */
    public String getStrPortalLogo() {
        return strPortalLogo;
    }

    /**
     * Sets the str portal logo.
     *
     * @param strPortalLogo
     *            the new str portal logo
     */
    public void setStrPortalLogo(final String strPortalLogo) {
        this.strPortalLogo = strPortalLogo;
    }

    /**
     * Gets the str portal css.
     *
     * @return the str portal css
     */
    public String getStrPortalCss() {
        return strPortalCss;
    }

    /**
     * Sets the str portal css.
     *
     * @param strPortalCss
     *            the new str portal css
     */
    public void setStrPortalCss(final String strPortalCss) {
        this.strPortalCss = strPortalCss;
    }

    /**
     * Gets the str created user.
     *
     * @return the str created user
     */
    public String getStrCreatedUser() {
        return strCreatedUser;
    }

    /**
     * Sets the str created user.
     *
     * @param strCreatedUser
     *            the new str created user
     */
    public void setStrCreatedUser(final String strCreatedUser) {
        this.strCreatedUser = strCreatedUser;
    }

    /**
     * Gets the str updated user.
     *
     * @return the str updated user
     */
    public String getStrUpdatedUser() {
        return strUpdatedUser;
    }

    /**
     * Sets the str updated user.
     *
     * @param strUpdatedUser
     *            the new str updated user
     */
    public void setStrUpdatedUser(final String strUpdatedUser) {
        this.strUpdatedUser = strUpdatedUser;
    }

    /**
     * Gets the created date.
     *
     * @return the created date
     */
    public Date getCreatedDate() {
        return createdDate;
    }

    /**
     * Sets the created date.
     *
     * @param createdDate
     *            the new created date
     */
    public void setCreatedDate(final Date createdDate) {
        this.createdDate = createdDate;
    }

    /**
     * Gets the updated date.
     *
     * @return the updated date
     */
    public Date getUpdatedDate() {
        return updatedDate;
    }

    /**
     * Sets the updated date.
     *
     * @param updatedDate
     *            the new updated date
     */
    public void setUpdatedDate(final Date updatedDate) {
        this.updatedDate = updatedDate;
    }

    /**
     * Gets the delete status.
     *
     * @return the delete status
     */
    public int getDeleteStatus() {
        return deleteStatus;
    }

    /**
     * Sets the delete status.
     *
     * @param deleteStatus
     *            the new delete status
     */
    public void setDeleteStatus(final int deleteStatus) {
        this.deleteStatus = deleteStatus;
    }

    /**
     * Gets the dashboards.
     *
     * @return the dashboards
     */
    public List<Dashboard> getDashboards() {
        return dashboards;
    }

    /**
     * Sets the dashboards.
     *
     * @param dashboards
     *            the new dashboards
     */
    public void setDashboards(final List<Dashboard> dashboards) {
        this.dashboards = dashboards;
    }

    /**
     * Checks if is portal view enabled.
     *
     * @return true, if is portal view enabled
     */
    public boolean isPortalViewEnabled() {
        return isPortalViewEnabled;
    }

    /**
     * Sets the portal view enabled.
     *
     * @param portalViewEnabled
     *            the new portal view enabled
     */
    public void setPortalViewEnabled(final boolean portalViewEnabled) {
        isPortalViewEnabled = portalViewEnabled;
    }
}
